File: Tecniques de Programació Avançades/Practica 2 - Aleix Carrillo i Oriol Poblet/src/main/java/RoundRobin.scala
import scala.util.control.Breaks._

class RoundRobin[U,V] extends PolicyManager[U,V] {
  private var j = 0

  override def valorsInvoker(invokers: List[Invoker[U, V]], requiredMemory: Int): Invoker[U, V] = {
    breakable {
      for (i <- invokers.indices) {
        val invoker = invokers(i)
        if (invoker.estaDisponible(requiredMemory)) {
          j = (j + 1) % invokers.size
          return invoker
        }
      }
    }
    throw new NoSuchElementException("No hi ha Invokers")
  }
}

File: Tecniques de Programació Avançades/Practica 1/src/RoundRobin.java
import java.util.List;
import java.util.NoSuchElementException;

public class RoundRobin implements PolicyManager {
    private int j;  

    /**
     * Constructor de la clase RoundRobin
     */

    public RoundRobin() {
        this.j = 0;
    }
    /**
     * Método que devuelve una lista de invokers que cumplen con los requisitos de memoria
     * @param invokers
     * @param requiredMemory
     * @return numInvokers lista de invokers en uso
     */

    @Override
    public Invoker valorsInvoker(List<Invoker> invokers, int requiredMemory) {

        for (int i = 0; i < invokers.size(); i++) {
            Invoker invoker = invokers.get(i);
            if (invoker.estaDisponible(requiredMemory)) {
                j = (j + 1) % invokers.size();
                return invoker;
            }
        }
        throw new NoSuchElementException("No hi ha Invokers");
    }
}
